lib/clerk/authenticate_context.rb

```diff
@@ -0,0 +1,186 @@
+        def initialize(request, config)
+            @clerk_url = URI.parse(request.url)
+            @config = config
+
+            @cookies = OpenStruct.new({
+                session_token_in_cookie: request.cookies[SESSION_COOKIE],
+                client_uat: request.cookies[CLIENT_UAT_COOKIE],
+                handshake_token: request.cookies[HANDSHAKE_COOKIE],
+                dev_browser: request.cookies[DEV_BROWSER_COOKIE]
+            })
+
+            @headers = OpenStruct.new({
+                session_token_in_header: request.env[AUTHORIZATION_HEADER].to_s.gsub(/bearer/i, '').strip,
+                sec_fetch_dest: request.env[SEC_FETCH_DEST_HEADER],
+                accept: request.env[ACCEPT_HEADER].to_s,
+                origin: request.env[ORIGIN_HEADER].to_s,
+                host: request.host,
+                port: request.port
+            })
+        end
+
+        ##
+        # The following properties are part of the props supported in all the AuthenticateContext
+        # objects across all of our SDKs (eg JS, Go)
+        def secret_key
+            @config.api_key.to_s
+        end
+
+        def publishable_key
+            @config.publishable_key.to_s
+        end
+
+        def domain
+            # TODO(dimkl): Add multi-domain support
+            ""
+        end
+
+        def is_satellite?
+            # TODO(dimkl): Add multi-domain support
+            false
+        end
+
+        def proxy_url
+            # TODO(dimkl): Add multi-domain support
+            ""
+        end
+
+        def handshake_token
+            @handshake_token ||= retrieve_from_query_string(@clerk_url, HANDSHAKE_COOKIE) || @cookies.handshake_token.to_s
+        end
+
+        def clerk_synced?
+            # TODO(dimkl): Add multi-domain support
+            false
+        end
+
+        def clerk_redirect_url
+             # TODO(dimkl): Add multi-domain support
+             ""
+        end
+
+        def dev_browser
+            @dev_browser ||= retrieve_from_query_string(@clerk_url, DEV_BROWSER_COOKIE) || @cookies.dev_browser.to_s
+        end
+
+        # The frontend_api returned is without protocol prefix
+        def frontend_api
+            return "" if !valid_publishable_key?(publishable_key.to_s)
+
+            @frontend_api ||= if !proxy_url.empty?
+                proxy_url
+            elsif development_instance? && !domain.empty?
+                "clerk.#{domain}"
+            else
+                # remove $ postfix
+                decode_publishable_key(publishable_key).chop
+            end
+        end
+
+        def development_instance?
+            # TODO(dimkl): Consolidate with @clerk/backend to use pk instead of sk
+            secret_key.start_with?("sk_test_")
+        end
+
+        def production_instance?
+            # TODO(dimkl): Consolidate with @clerk/backend to use pk instead of sk
+            secret_key.start_with?("sk_live_")
```

Comment:
These may also start with `live_`

Reply:
After discussing it with @SokratisVidros we considered dropping the legacy AuthV1 format. Also since we are gonna do a major bump for this change, i think we are okay. Does it make any sense? Do you have any considerations?